src/cloudflare/internal/secrets-store-api.ts

```diff
@@ -0,0 +1,32 @@
+type Fetcher = {
+  fetch: typeof fetch;
+};
+
+export default function makeBinding(env: { fetcher: Fetcher }): SecretsStore {
+  return new SecretsStoreBindingImpl(env.fetcher);
+}
+
+type SecretsStoreProxyWorkerResponse = {
+  name: string;
+  value: string;
+};
+
+class SecretsStoreBindingImpl implements SecretsStore {
+  public constructor(private readonly fetcher: Fetcher) {}
+
+  public async get(): Promise<string> {
+    try {
+      const response = await this.fetcher.fetch(
```

Comment:
i believe we'd need something like this example binding here (without the fetchOptions) 
```
const endpointUrl =
      'https://workers-binding.ai/to-everything/markdown/transformer';

    const res = await this.fetcher.fetch(endpointUrl, fetchOptions);
    ```